test/e2e/nodepool_upgrade_test.go

```diff
@@ -47,6 +51,116 @@ func TestUpgradeNodePool(t *testing.T) {
 		}
 	}
 
+	// Look up metadata about the release images so that we can extract the version
+	// information for later assertions.
+	releaseInfoProvider := &releaseinfo.RegistryClientProvider{}
+	pullSecretFile, err := os.Open(clusterOpts.PullSecretFile)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to open pull secret file")
+	defer pullSecretFile.Close()
+	pullSecret, err := ioutil.ReadAll(pullSecretFile)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to read pull secret file")
+	previousReleaseInfo, err := releaseInfoProvider.Lookup(ctx, globalOpts.PreviousReleaseImage, pullSecret)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get release info for previous image")
+	latestReleaseInfo, err := releaseInfoProvider.Lookup(ctx, globalOpts.LatestReleaseImage, pullSecret)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get release info for latest image")
+
+	// Create the test cluster.
+	hostedCluster := e2eutil.CreateCluster(t, ctx, client, &clusterOpts, hyperv1.AWSPlatform, globalOpts.ArtifactDir)
+
+	// Wait for connectivity to the cluster.
+	t.Logf("Waiting for guest client to become available")
+	guestClient := e2eutil.WaitForGuestClient(t, ctx, client, hostedCluster)
+
+	// Wait for Nodes to be Ready.
+	numNodes := int32(globalOpts.configurableClusterOptions.NodePoolReplicas * len(clusterOpts.AWSPlatform.Zones))
+	e2eutil.WaitForNReadyNodes(t, ctx, guestClient, numNodes)
+
+	// Wait for the first rollout to be complete and refresh the HostedCluster.
+	t.Logf("Waiting for initial cluster rollout. Image: %s", hostedCluster.Spec.Release.Image)
+	e2eutil.WaitForImageRollout(t, ctx, client, hostedCluster, hostedCluster.Spec.Release.Image)
+	err = client.Get(ctx, crclient.ObjectKeyFromObject(hostedCluster), hostedCluster)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get hostedcluster")
+
+	// Find NodePools.
+	var nodePools hyperv1.NodePoolList
+	err = client.List(ctx, &nodePools, &crclient.ListOptions{Namespace: hostedCluster.Namespace})
+	g.Expect(err).NotTo(HaveOccurred(), "failed to list NodePools")
+
+	// Wait for NodePools to roll out the initial version.
+	// TODO: Consider doing this in parallel
+	for _, nodePool := range nodePools.Items {
+		e2eutil.WaitForNodePoolVersion(t, ctx, client, &nodePool, previousReleaseInfo.Version())
+	}
+
+	// Update NodePool images to the latest.
+	for _, nodePool := range nodePools.Items {
+		err = client.Get(ctx, crclient.ObjectKeyFromObject(&nodePool), &nodePool)
+		g.Expect(err).NotTo(HaveOccurred(), "failed to get NodePool")
+		t.Logf("Updating NodePool image. Image: %s", globalOpts.LatestReleaseImage)
+		nodePool.Spec.Release.Image = globalOpts.LatestReleaseImage
```

Comment:
Hmm, so generally how much does this differ by? Could they be the same image?

Alternatively, if the release image are close enough, we could have the same MCO templates + same OS image, in which case (once we have feature parity with the self-driving MCO), there will be no update (and the monitor below for Unready node will not trigger, since the MCO sees that no update is needed)